jdaviz/app.py

```diff
@@ -828,7 +828,7 @@ def get_subsets_from_viewer(self, viewer_reference, data_label=None, subset_type
         return regions
 
     def get_subsets(self, subset_name=None, spectral_only=False,
-                    spatial_only=False, object_only=False):
+                    spatial_only=False, object_only=False, simplify_spectral=True):
```

Comment:
I think we'll eventually want this to default to `False` (or perhaps not have the kwarg at all and have a separate `simplify_spectral_subsets()` method which _applies_ the change directly to the glue state).  Is that the eventual goal or do others have different thoughts on the eventual direction?

This may still be the simplest incremental step towards that direction though and if so, is ok with me for now.

Reply:
@camipacifici mentioned that users would prefer the simplified version since it makes sense with what they're seeing visually. I can only see the non-simplified version being useful for Subset Plugin, although even there we may decide to stick with the simplified version. The reason being that there are multiple states (i.e. subset_state.state1, state1.state2, state1.state2.state1, etc.) that are duplicates and editing one may not reflect the changes in the viewer.

Reply:
ok, then `simplify_spectral` could call the method first before exposing the information in the same dictionary format?  I just think it would be nice to have a consistent output-format (between spectral and spatial subsets) for default kwargs.  Right now, the default returned object is quite different between spectral and spatial 🤷 .  I also worry if we are going to show the actual state in the plugin by default, but the simplified representation in the API, that that could cause some confusion for the user.

Reply:
I think in the event of confusion we should default to the simpler version. Is the main reason to keep the non-simplified version consistency with the spatial subset? They are different things so I think it makes sense to represent them differently.

Reply:
Right, but it is the same _method_ and the output formatting changes based on the "context" (rather than user-inputs).  In other words `imviz.app.get_subsets()` returns these dictionary structures (non-simplified), whereas `specviz.app.get_subsets()` returns a completely different type of object and which is now simplified and doesn't match what we'll eventually want to show in the plugin.  I just think this is difficult for an API where users might want to parse the returned object, since they'll need to add custom logic to determine how to parse it. 

But if this is staying within the app-level, we could also address this issue when exposing it to the helper or plugin API.

Reply:
My understanding is this is staying in the app-level and certain features can be exposed by helper methods.

Reply:
if its staying in app-level, then maybe the argument of what users would find most convenient doesn't apply yet 😉 .  Ok, I won't hold up the PR for this and we can revisit later, but I am concerned that changing output formats will be a headache as we start to use this as well.

Reply:
I think I'm less concerned with _changing_ the output format with this argument, than I am with the fact that the format within the same output dictionary is _mixed_ in the Cubeviz case when you have both spatial and spectral subsets defined. The first solution that comes to mind is returning only the spectral subsets if `simplify_spectral` is `True`, but that would necessitate changing the default to `False` so that you always get everything back by default.